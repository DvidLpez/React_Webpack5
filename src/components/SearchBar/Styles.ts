import styled from "styled-components";
import {
  Wrapper,
} from "../../components/StyledComponents/styledComponents";

export const WrapperSearch = styled(Wrapper)`
  position: relative;
  padding: ${(props) => props.theme.spaces.s30};

  @media screen and (min-width: 768px) {
    padding: 10px 30px;
  }
`;
export const WrapperStates = styled(Wrapper)`
  text-align: ${(props) => props.theme.spaces.center};
  label {
    font-size: ${(props) => props.theme.fonts.medium};
    font-weight: ${(props) => props.theme.fonts.bold};
    margin: 0 6%;
    cursor: pointer;
  }
  label.open {
    color: ${(props) => props.theme.colors.success};
  }
  label.closed {
    color: ${(props) => props.theme.colors.closed};
  }

  @media screen and (min-width: 768px) {
    padding: ${(props) => props.theme.spaces.s15};
    label {
      font-size: ${(props) => props.theme.fonts.large};
    }
  }
`;



export const InputSearch = styled.input`
  border: 1px solid #838383;
  border-radius: ${(props) => props.theme.spaces.s30};
  color: #2e2e2e;
  font-size: ${(props) => props.theme.fonts.default};
  font-style: italic;
  outline: none;
  padding: 7px 10px 7px 10px;
  width: 100%;
  -webkit-box-shadow: 0 0 20px 5px rgb(163, 163, 163);
  -moz-box-shadow: 0 0 20px 5px rgba(163, 163, 163, 1);
  box-shadow: 0 0 20px 5px rgba(163, 163, 163, 1);
  text-indent: 55px;
  transition: all 0.3s ease-in-out;

  &::-webkit-input-placeholder {
    color: #cccccc;
    text-indent: 50px;
    font-weight: 300;
  }

  &:focus,
  &:active {
    text-indent: ${(props) => props.theme.spaces.none};
    background: #fff;
    &::-webkit-input-placeholder {
      color: #aaa;
      text-indent: 3px;
    }
    + label {
      transform: rotate(-66deg);
      border-radius: 30px;

      &:before {
        transform: rotate(10deg);
      }
    }
  }

  + label {
    display: inline-block;
    position: absolute;
    top: 30px;
    left: 30px;
    width: 65px;
    height: 50px;
    font-size: ${(props) => props.theme.fonts.medium};
    padding: 10px 15px;
    text-shadow: 0 1px 0 rgba(19, 74, 70, 0.4);
    color: white;

    transition: all 0.4s ease-in-out;
    border-radius: 30px;
    transform-origin: left bottom;
    z-index: 99;

    &:before,
    &:after {
      content: "";
      position: absolute;
      top: ${(props) => props.theme.spaces.none};
      right: ${(props) => props.theme.spaces.none};
      bottom: ${(props) => props.theme.spaces.none};
      left: ${(props) => props.theme.spaces.none};
      width: 55px;
      height: 35px;
      border-radius: 30px;
      border-top-right-radius: ${(props) => props.theme.spaces.none};
      border-bottom-right-radius: ${(props) => props.theme.spaces.none};
      background: #377d6a;
      transform-origin: left bottom;
      transition: all 0.4s ease-in-out;
      pointer-events: none;
      z-index: -1;
    }

    &:before {
      background: rgba(3, 36, 41, 0.2);
      z-index: -2;
      right: 20%;
    }
  }

  @media screen and (min-width: 768px) {
    font-size: ${(props) => props.theme.fonts.medium};
    padding: 15px 20px 15px 40px;
    text-indent: 90px;
    &::-webkit-input-placeholder {
      text-indent: 90px;
    }
    + label {
      top: 9px;
      left: 30px;
      width: 65px;
      height: 50px;
      font-size: ${(props) => props.theme.fonts.medium};
      padding: 10px 15px;
      transition: all 0.4s ease-in-out;
      transform-origin: left bottom;
      &:before,
      &:after {
        width: 100px;
        height: 55px;
      }
    }
  }
`;

