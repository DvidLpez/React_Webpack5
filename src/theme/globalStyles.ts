
import { createGlobalStyle } from "styled-components";
import { theme } from "./theme";


export const GlobalStyle = createGlobalStyle`

   html {
      box-sizing: border-box;
   }
   body {
         background-color: ${theme.colors.tertiary};
         font-family: ${theme.fonts.style};
         padding: ${theme.spaces.none};
         margin: ${theme.spaces.none};
   }
   *,
   ::after,
   ::before {
      box-sizing: inherit;
   }

   a{
      text-decoration: none;
      color: ${theme.colors.primary};
   }

   hr{
      border: none;
      border-bottom: solid 1px ${theme.colors.secondary};
   }

   .icon_brand {
      font-size: ${theme.fonts.big};
      margin-right: ${theme.spaces.s15};
      color: ${theme.colors.title};
   }
   .icon_sweat {
      font-size: ${theme.fonts.extra};
   }

   .icon_error {
         font-size: ${theme.fonts.extraBig};
      }

   @media screen and (min-width: 768px) {
   
      .icon_brand {
         font-size: ${theme.fonts.extra};
      }
      
   }
`;
