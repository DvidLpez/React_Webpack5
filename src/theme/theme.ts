import { DefaultTheme } from "styled-components";

export const theme: DefaultTheme = {
  fonts: {
    style: "'Source Sans Pro', sans-serif",
    small: "0.9em",
    default: "1em",
    medium: "1.2em",
    large: "1.6em",
    big: "2em",
    extra: "3em",
    extraBig: "5em",
    bold: "600",
  },
  colors: {
    primary: "#000",
    secondary: "#dbdbdb",
    tertiary: "#fafafa",
    grey: "#414141",
    lightgrey: "#7c7c7c",
    gradient:
      "linear-gradient(to right, #3e9e7e 0%, #2e755d 10%, #205040 20%, #414141 30%)",
    title: "#fff",
    success: "#008000",
    closed: "#c60404",
    danger: "#932323",
    backComment: "#fff9d6",
    shadow: "2px 4px 11px 1px rgba(138, 138, 138, 1)",
  },
  spaces: {
    none: "0",
    center: "center",
    s5: "5px",
    s10: "10px",
    s15: "15px",
    s20: "20px",
    s25: "25px",
    s30: "30px",
    s50: "50px",
    p55: "5px 5px",
    p100: "10px 0",
    p1520: "15px 20px",
    p1010: "10px 10px",
    p1515: "15px 15px",
    p2020: "20px 20px",
    p2525: "25px 25px",
    p3030: "30px 30px",
    button: "10px 40px",
    borderRadius: "5px",
  }
};
