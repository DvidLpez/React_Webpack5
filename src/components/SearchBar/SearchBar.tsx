import React, { FC, Fragment, ReactElement, useEffect} from "react";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch } from '@fortawesome/free-solid-svg-icons';
import { InputSearch, WrapperSearch, WrapperStates } from './Styles';
import { inputVar } from '../../cache';
import { LoadIssues } from '../../graphql/manage';
import { useReactiveVar } from "@apollo/client";


const SearchBar: FC = (): ReactElement => {

  
   const { term, status } = useReactiveVar(inputVar);
   
   const handleForm = (e:React.ChangeEvent<HTMLInputElement>) => { 
      e.target.name === "term" 
         ? inputVar({term: e.target.value, status}) 
         : inputVar({term, status: e.target.value}); 
   }


   useEffect( ()=> {

      console.log('Dentro de effect term y status - ' , inputVar());
      const delayDebounce = setTimeout(() => {
            if (term.trim().length > 2) {
               LoadIssues(term.trim(), status, 9, 'next', null);
            }
            return () => clearTimeout(delayDebounce);
      }, 500);
   }, [term, status]);

   return(
      <Fragment>
         <WrapperSearch>
               <InputSearch
                  id="searchbar"
                  type="text"
                  name="term"
                  placeholder="Search term"
                  onChange={handleForm}
                  value={term}
               />
               <label><FontAwesomeIcon icon={faSearch} className="icon_search" /></label>
         </WrapperSearch>
         <WrapperStates>
            <label>ALL
               <input
                  id="all"
                  type="radio"
                  name="state"
                  onChange={handleForm}
                  value=""
                  checked={'' == status ? true : false}
               />
            </label>
            <label className="open">OPEN
               <input 
                  id="open"
                  type="radio"
                  name="state"
                  onChange={handleForm}
                  value="open"
                  checked={'open' == status ? true : false}
               />
            </label>
            <label className="closed">CLOSED
               <input 
                  id="closed"
                  type="radio"
                  name="state"
                  onChange={handleForm}
                  value="closed"
                  checked={'closed' == status ? true : false}
               />
            </label>
         </WrapperStates>
      </Fragment>
   );
}
export default SearchBar;
