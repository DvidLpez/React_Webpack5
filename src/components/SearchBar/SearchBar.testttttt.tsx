import React from "react";
import { mount, shallow, render } from 'enzyme';
import { ThemeProvider } from "styled-components";
import { theme } from "../../theme/theme";
import SearchBar from "./SearchBar";

export const mountWithTheme = (children: any) =>
   mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>);

export const renderWithTheme = (children: any) =>
   render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);

export const shallowWithTheme = (children: any) =>
   shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>);



describe('Testing SearchBar', () => {



   // it('renders snapshots, too', () => {
   //    const wrapper = shallow(<SearchBar />)
   //    expect(wrapper).toMatchSnapshot()
   // })

})