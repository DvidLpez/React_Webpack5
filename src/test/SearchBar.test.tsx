import React from "react";
import { mount, shallow, render } from 'enzyme';
import { ThemeProvider } from "styled-components";
import { theme } from "../theme/theme";
import SearchBar from "../components/SearchBar/SearchBar";
import { Provider } from 'react-redux';
import store from "../redux/store";
import * as reactRedux from 'react-redux';

const renderWrapper = ()=> {
   return render(
      <Provider store={store}>
         <ThemeProvider theme={theme}>
            <SearchBar />
         </ThemeProvider>
      </Provider>
   )
}

const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');

describe('TESTING SEARCHBAR COMPONENT', () => {
   
   it('- Default Searchbar HTML output', () => {
      beforeEach(() => {
         useSelectorMock.mockClear();
      });
      useSelectorMock.mockReturnValue({term: '', status: ''});
      const wrapper = renderWrapper(); 
      expect(wrapper.find('#all').prop('checked')).toBe(true); 
      expect(wrapper.find('#open').prop('checked')).toBe(false); 
      expect(wrapper.find('#closed').prop('checked')).toBe(false); 
      expect(wrapper.find('#searchbar').val()).toMatch(''); 
      expect(wrapper.find('input')).toHaveLength(4); 
      expect(wrapper).toMatchSnapshot(); 
   });

   describe('- CHECK STATES OPEN AND CLOSE', () => {
      it('- Input open is checked', () => {
         beforeEach(() => {
            useSelectorMock.mockClear();
         });
         useSelectorMock.mockReturnValue({ term: '', status: 'open' });
         const wrapper = renderWrapper();
         expect(wrapper.find('#all').prop('checked')).toBe(false); 
         expect(wrapper.find('#open').prop('checked')).toBe(true); 
         expect(wrapper.find('#closed').prop('checked')).toBe(false); 
         expect(wrapper.find('input')).toMatchSnapshot(); 
      });

      it('- Input closed is checked', () => {
         beforeEach(() => {
            useSelectorMock.mockClear();
         });
         useSelectorMock.mockReturnValue({ term: '', status: 'closed' });
         const wrapper = renderWrapper();
         expect(wrapper.find('#all').prop('checked')).toBe(false); 
         expect(wrapper.find('#open').prop('checked')).toBe(false); 
         expect(wrapper.find('#closed').prop('checked')).toBe(true); 
         expect(wrapper.find('input')).toMatchSnapshot(); 
      });
   });

   describe('- CHECK SEARCH VALUE', () => {
      it('- Add vaule as a term to search', () => {
         beforeEach(() => {
            useSelectorMock.mockClear();
         });
         useSelectorMock.mockReturnValue({ term: 'useEffect', status: '' });
         const wrapper = renderWrapper();
         expect(wrapper.find('#searchbar').val()).toMatch('useEffect');
         expect(wrapper.find('#searchbar')).toMatchSnapshot();
      });
   });
   
});
