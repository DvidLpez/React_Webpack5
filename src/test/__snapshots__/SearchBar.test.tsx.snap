// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTING SEARCHBAR COMPONENT - CHECK SEARCH VALUE - Add vaule as a term to search 1`] = `
<input
  class="sc-jrAGrp fCthTV"
  id="searchbar"
  name="term"
  placeholder="Search term"
  type="text"
  value="useEffect"
/>
`;

exports[`TESTING SEARCHBAR COMPONENT - CHECK STATES OPEN AND CLOSE - Input closed is checked 1`] = `
Array [
  <input
    class="sc-jrAGrp fCthTV"
    id="searchbar"
    name="term"
    placeholder="Search term"
    type="text"
    value=""
  />,
  <input
    id="all"
    name="state"
    type="radio"
    value=""
  />,
  <input
    id="open"
    name="state"
    type="radio"
    value="open"
  />,
  <input
    checked=""
    id="closed"
    name="state"
    type="radio"
    value="closed"
  />,
]
`;

exports[`TESTING SEARCHBAR COMPONENT - CHECK STATES OPEN AND CLOSE - Input open is checked 1`] = `
Array [
  <input
    class="sc-jrAGrp fCthTV"
    id="searchbar"
    name="term"
    placeholder="Search term"
    type="text"
    value=""
  />,
  <input
    id="all"
    name="state"
    type="radio"
    value=""
  />,
  <input
    checked=""
    id="open"
    name="state"
    type="radio"
    value="open"
  />,
  <input
    id="closed"
    name="state"
    type="radio"
    value="closed"
  />,
]
`;

exports[`TESTING SEARCHBAR COMPONENT - Default Searchbar HTML output 1`] = `
Array [
  <div
    class="sc-bdfBwQ sc-fubCfw kUdDos jYqrAh"
  >
    <input
      class="sc-jrAGrp fCthTV"
      id="searchbar"
      name="term"
      placeholder="Search term"
      type="text"
      value=""
    />
    <label>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-search fa-w-16 icon_search"
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
        />
      </svg>
    </label>
  </div>,
  <div
    class="sc-bdfBwQ sc-pFZIQ kUdDos ckbQdV"
  >
    <label>
      ALL
      <input
        checked=""
        id="all"
        name="state"
        type="radio"
        value=""
      />
    </label>
    <label
      class="open"
    >
      OPEN
      <input
        id="open"
        name="state"
        type="radio"
        value="open"
      />
    </label>
    <label
      class="closed"
    >
      CLOSED
      <input
        id="closed"
        name="state"
        type="radio"
        value="closed"
      />
    </label>
  </div>,
]
`;
